### ISSUE TITLE
Plugin: Bookmarks for maps and portals (bookmarks-by-zaso)

### ISSUE THREAD

**Issue by Novoca1n3** [jonatkins/ingress-intel-total-conversion#1170]
Any possibility of adding the ability to import KML files? Similar to the overlay KML plug-in however they are stored as bookmarks instead of an overlay

**Comment by nhamer** [comment]
You should probably contact Zaso directly (he's a friendly type) at https://github.com/ZasoGD

**Comment by whartyENL** [comment]
Thanks

Get Outlook for Android<https://aka.ms/ghei36>



On Fri, Jan 6, 2017 at 6:34 PM +1100, "nhamer" <notifications@github.com<mailto:notifications@github.com>> wrote:


You should probably contact Zaso directly (he's a friendly type) at https://github.com/ZasoGD

**Comment by nhamer** [comment]
@zasogd may correct me, but thinking about it - imported bookmarks need portal GUIDs, which you aren't going to find in KML.

**Comment by whartyENL** [comment]
I can import KML now and have it put location markers on portals therefore all the required information must already exist. Attached is an example KML file

Get Outlook for Android<https://aka.ms/ghei36>




On Sat, Jan 21, 2017 at 3:47 PM +1100, "nhamer" <notifications@github.com<mailto:notifications@github.com>> wrote:


@zasogd<https://github.com/zasogd> may correct me, but thinking about it - imported bookmarks need portal GUIDs, which you aren't going to find in KML.

**Comment by nhamer** [comment]
Can you upload the kml either directly here to github or to pastebin or
something? It didn't come through in mail.

**Comment by whartyENL** [comment]
Silly question, how do I add attachments :-)

**Comment by nhamer** [comment]
Oh. I suppose the answer is actually "upload to pastebin, add a link". Sorry, just getting used to the toolchain.

**Comment by ZasoGD** [comment]
In _IITC_ A portal can be found without using its _GUID_ [`window.selectPortalByLatLng(lat, lng)`](https://github.com/iitc-project/ingress-intel-total-conversion/blob/master/code/utils_misc.js#L271).

If you have not the _GUID_ is not a problem (you can create a fake _GUID_ when you add/create one o more portal(s) in _bookmarks_), but _bookmarks_ when you click a portal bookmark from the list uses the [`window.zoomToAndShowPortal(guid, latlng)`](https://github.com/iitc-project/ingress-intel-total-conversion/blob/master/plugins/bookmarks-by-zaso.user.js#L263) to move the map and select the portal. 
Furthermore, without the _GUIDs_ the _star_ not works properly.

It's not possible import correctly _KML_ data in _bookmarks_ if in the _KML_ are not the _GUID_ of portals. 

Why don't edit the plugin? I'll open an issue about this (I hope that @nhamer and others will want to partecipate).

**Comment by whartyENL** [comment]
Here's an example KML file created by exporting from google maps. It contains ingress portals for overlay in IITC http://pastebin.com/rPiEAaGk

**Comment by nhamer** [comment]
Yeah, no GUIDs. So essentially, highlight can't work, star can't work. I suppose you could create a fake GUID on import, but the limitation is going to be confusing. 

What's your use case here? It would seem more natural to import KML to e.g. drawtools.

**Comment by whartyENL** [comment]
These kml files overlay portals in IITC (using the Overlay KML plugin) making it easy to locate and navigate to them. The problem is that when iitc is closed, you need to load the file again. Saving as bookmarks means they would stay loaded and be synced between devices

**Comment by nhamer** [comment]
Ref. #1154 possibly - possibly overlay-kml should persist a kml and handle network-links? I wasn't aware it didn't even save the KML between sessions (that's dum), but network links would also keep it current (not even using sync plugins)?

**Comment by whartyENL** [comment]
If you're sharing files between people I think you may need a shared network location to link too. Not sure as I've never tried it
